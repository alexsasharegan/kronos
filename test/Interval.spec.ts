import {
  Millisecond,
  Day,
  Hour,
  Microsecond,
  Minute,
  Second,
  Week,
} from "../src/Interval";

const TruthTable = {
  Microsecond: {
    Microsecond: 1,
    Millisecond: 1 / 1000,
    Second: 1 / (1000 * 1000),
    Minute: 1 / (60 * 1000 * 1000),
    Hour: 1 / (60 * 60 * 1000 * 1000),
    Day: 1 / (24 * 60 * 60 * 1000 * 1000),
    Week: 1 / (7 * 24 * 60 * 60 * 1000 * 1000),
  },
  Millisecond: {
    Microsecond: 1000,
    Millisecond: 1,
    Second: 1 / 1000,
    Minute: 1 / (60 * 1000),
    Hour: 1 / (60 * 60 * 1000),
    Day: 1 / (24 * 60 * 60 * 1000),
    Week: 1 / (7 * 24 * 60 * 60 * 1000),
  },
  Second: {
    Microsecond: 1 * 1000 * 1000,
    Millisecond: 1 * 1000,
    Second: 1,
    Minute: 1 / 60,
    Hour: 1 / (60 * 60),
    Day: 1 / (24 * 60 * 60),
    Week: 1 / (7 * 24 * 60 * 60),
  },
  Minute: {
    Microsecond: 1 * 60 * 1000 * 1000,
    Millisecond: 1 * 60 * 1000,
    Second: 1 * 60,
    Minute: 1,
    Hour: 1 / 60,
    Day: 1 / (24 * 60),
    Week: 1 / (7 * 24 * 60),
  },
  Hour: {
    Microsecond: 1 * 60 * 60 * 1000 * 1000,
    Millisecond: 1 * 60 * 60 * 1000,
    Second: 1 * 60 * 60,
    Minute: 1 * 60,
    Hour: 1,
    Day: 1 / 24,
    Week: 1 / (7 * 24),
  },
  Day: {
    Microsecond: 1 * 24 * 60 * 60 * 1000 * 1000,
    Millisecond: 1 * 24 * 60 * 60 * 1000,
    Second: 1 * 24 * 60 * 60,
    Minute: 1 * 24 * 60,
    Hour: 1 * 24,
    Day: 1,
    Week: 1 / 7,
  },
  Week: {
    Microsecond: 1 * 7 * 24 * 60 * 60 * 1000 * 1000,
    Millisecond: 1 * 7 * 24 * 60 * 60 * 1000,
    Second: 1 * 7 * 24 * 60 * 60,
    Minute: 1 * 7 * 24 * 60,
    Hour: 1 * 7 * 24,
    Day: 1 * 7,
    Week: 1,
  },
} as const;

const RootTestTable = [
  {
    label: "Microsecond",
    interval: Microsecond(1),
    table: TruthTable.Microsecond,
  },
  {
    label: "Millisecond",
    interval: Millisecond(1),
    table: TruthTable.Millisecond,
  },
  {
    label: "Second",
    interval: Second(1),
    table: TruthTable.Second,
  },
  {
    label: "Minute",
    interval: Minute(1),
    table: TruthTable.Minute,
  },
  {
    label: "Hour",
    interval: Hour(1),
    table: TruthTable.Hour,
  },
  {
    label: "Day",
    interval: Day(1),
    table: TruthTable.Day,
  },
  {
    label: "Week",
    interval: Week(1),
    table: TruthTable.Week,
  },
];

for (const { interval, table, label } of RootTestTable) {
  describe(label, () => {
    const tt = [
      { actual: interval.toMicroseconds(), expected: table.Microsecond },
      { actual: interval.toMilliseconds(), expected: table.Millisecond },
      { actual: interval.toSeconds(), expected: table.Second },
      { actual: interval.toMinutes(), expected: table.Minute },
      { actual: interval.toHours(), expected: table.Hour },
      { actual: interval.toDays(), expected: table.Day },
      { actual: interval.toWeeks(), expected: table.Week },
    ];

    for (const { actual, expected } of tt) {
      test(`${actual.toString({ mode: "verbose" })}`, () => {
        expect(actual.value).toBe(expected);
      });
    }
  });
}
