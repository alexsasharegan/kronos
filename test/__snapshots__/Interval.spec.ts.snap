// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day Day#toString: Day abbreviated 1`] = `"1d"`;

exports[`Day Day#toString: Day default 1`] = `"1 day"`;

exports[`Day Day#toString: Day verbose 1`] = `"1 day"`;

exports[`Hour Hour#toString: Hour abbreviated 1`] = `"1hr"`;

exports[`Hour Hour#toString: Hour default 1`] = `"1 hour"`;

exports[`Hour Hour#toString: Hour verbose 1`] = `"1 hour"`;

exports[`Microsecond Microsecond#toString: Microsecond abbreviated 1`] = `"1Âµs"`;

exports[`Microsecond Microsecond#toString: Microsecond default 1`] = `"1 microsecond"`;

exports[`Microsecond Microsecond#toString: Microsecond verbose 1`] = `"1 microsecond"`;

exports[`Millisecond Millisecond#toString: Millisecond abbreviated 1`] = `"1ms"`;

exports[`Millisecond Millisecond#toString: Millisecond default 1`] = `"1 millisecond"`;

exports[`Millisecond Millisecond#toString: Millisecond verbose 1`] = `"1 millisecond"`;

exports[`Minute Minute#toString: Minute abbreviated 1`] = `"1min"`;

exports[`Minute Minute#toString: Minute default 1`] = `"1 minute"`;

exports[`Minute Minute#toString: Minute verbose 1`] = `"1 minute"`;

exports[`Second Second#toString: Second abbreviated 1`] = `"1sec"`;

exports[`Second Second#toString: Second default 1`] = `"1 second"`;

exports[`Second Second#toString: Second verbose 1`] = `"1 second"`;

exports[`Week Week#toString: Week abbreviated 1`] = `"1wk"`;

exports[`Week Week#toString: Week default 1`] = `"1 week"`;

exports[`Week Week#toString: Week verbose 1`] = `"1 week"`;
